ui/pages/bridge/prepare/bridge-cta-button.tsx

```diff
@@ -108,20 +109,29 @@ export const BridgeCTAButton = () => {
       data-testid="bridge-cta-button"
       onClick={() => {
         if (activeQuote && isTxSubmittable) {
-          quoteRequestProperties &&
-            requestMetadataProperties &&
-            tradeProperties &&
-            trackCrossChainSwapsEvent({
-              event: MetaMetricsEventName.ActionSubmitted,
-              properties: {
-                ...quoteRequestProperties,
-                ...requestMetadataProperties,
-                ...tradeProperties,
-              },
-            });
-          submitBridgeTransaction(activeQuote);
+          try {
+            // We don't need to worry about setting to true if the tx submission succeeds
+            // because we route immediately to Activity list page
+            setIsSubmitting(true);
+
+            quoteRequestProperties &&
+              requestMetadataProperties &&
+              tradeProperties &&
+              trackCrossChainSwapsEvent({
+                event: MetaMetricsEventName.ActionSubmitted,
+                properties: {
+                  ...quoteRequestProperties,
+                  ...requestMetadataProperties,
+                  ...tradeProperties,
+                },
+              });
+            submitBridgeTransaction(activeQuote);
+          } catch (error) {
+            setIsSubmitting(false);
+          }
```

Comment:
Can we use `finally` instead?

Reply:
Handled in https://github.com/MetaMask/metamask-extension/pull/29149